hare/algorithm.go

```diff
@@ -45,21 +45,36 @@ type Signer interface {
 	PublicKey() *signing.PublicKey
 }
 
-// procOutput is the tuple (id, set) which is the output of the consensus process.
-type procOutput struct {
-	id  instanceId
-	set *Set
+const (
+	completed    = true
+	notCompleted = false
+)
+
+// procReport is the termination report of the CP.
+// It consistes of the layer id, the set we agreed on (if available) and a flag to indicate if the CP completed.
+type procReport struct {
+	id        instanceId
+	set       *Set
+	completed bool
 }
 
-func (cpo procOutput) Id() instanceId {
+func (cpo procReport) Id() instanceId {
 	return cpo.id
 }
 
-func (cpo procOutput) Set() *Set {
+func (cpo procReport) Set() *Set {
 	return cpo.set
 }
 
-var _ TerminationOutput = (*procOutput)(nil)
+func (cpo procReport) Completed() bool {
+	return cpo.completed
+}
+
+func (proc *ConsensusProcess) report(completed bool) {
+	proc.terminationReport <- procReport{proc.instanceId, proc.s, completed}
+}
+
+var _ TerminationOutput = (*procReport)(nil)
```

Comment:
This statement is redundant. Since `proc.terminationReport` the line above (74) would fail if the interface wasn't implemented by `procReport `.

Reply:
That is what this line comes to check @y0sher